### PR TITLE
Admin becoming user password hint#168385051

### PR DESCRIPTION
**Pull request by BasilMawejje**
## Changes proposed in this pull request:
1.  When an admin becomes a user and clicks 'Edit your profile' in the header dropdown menu, a pop up should come to the screen to remind the admin to use the admin password as the current password. This should happen when the admin sets focus to the current password field. 

2. If the admin uses the tab key to navigate to the current password field, the pop up should still be activated to remind the admin which password to use. 

## Screenshots (Optional):
Screenshot 2019-09-13 at 15 17 34

---
## Ready for review:
@patmbolger @thesuss @weedySeaDragon

### CODE REVIEW

File: features/step_definitions/admin-becoming-user-password-hint_steps.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Given("I am on the {capture_string}{optional_string}{optional_string}{capture_string}") do |capture_string, optional_string, optional_string2, capture_string2|
+  visit admin_only_user_profile_edit_path(@user)
+end
+
+Then("I should navigate to {capture_string}") do |capture_string|
```
[the remark this answers is not in the document]
Comment (weedySeaDragon) [reply features/step_definitions/admin-becoming-user-password-hint_steps.rb]: Hound can't parse this correctly; we don't need to do anything about this.

**Comment by weedySeaDragon** [comment]
@patmbolger  - While I was writing cucumber tests for @BasilMawejje, it seemed to me that we have no way to tell if the current user was actually the admin that _became_ this user.

The [devise_masquerade  gem](https://github.com/oivoodoo/devise_masquerade) does all of this: allows you to 'become' ("masquerade as") a user and keeps track if the current user was an admin that 'became' this user

It has a simple method that we can use to see if the current user has been 'masqueraded':
`user_masquerade? # current user has been masqueraded (our "admin has 'become' a user")`

Instead of writing (and maintaining) our own functionality to do keep track of if an admin has 'become' a user, I think that we should use the `devise_masquerade` gem to accomplish this.  It's a well known gem that has been used for years. 

Don't know if we should create a new PT story and PR or what.  

But we should all discuss this -- here and/or during the next team meeting.

**Comment by BasilMawejje** [comment]
@weedySeaDragon In the spec background I made both users admins in order to work around the Pundit::NotDefinedError that comes when an admin becomes a non-admin user in the tests. All tests are green. You can review and give me feedback on the implementation.

**Comment by weedySeaDragon** [comment]
The system is already doing what it should

Once the admin "becomes"  a user, the system expects that the password entered is the _user's_ password.

Hence we discovered we don't need this PR and are closing it.

**Pull request closed without merging**